pkg/oci/builder_test.go

```diff
@@ -46,7 +46,69 @@ func TestBuilder_Build(t *testing.T) {
 
 	last := path(f.Root, fn.RunDataDir, "builds", "last", "oci")
 
-	validateOCI(last, t)
+	validateOCIStructure(last, t) // validate it adheres to the basics of the OCI spec
+}
+
+// TestBuilder_Files ensures that static files are added to the container
+// image as expected.  This includes template files, regular files and links.
+func TestBuilder_Files(t *testing.T) {
+	root, done := Mktemp(t)
+	defer done()
+
+	// Create a function with the default template
+	f, err := fn.New().Init(fn.Function{Root: root, Runtime: "go"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Add a regular file
+	if err := os.WriteFile("a.txt", []byte("file a"), 0644); err != nil {
+		t.Fatal(err)
+	}
+
+	// Links
+	var link struct {
+		Target     string
+		Mode       fs.FileMode
+		Executable bool
+	}
+	if runtime.GOOS != "windows" {
+		// Default case: use symlinks
+		link.Target = "a.txt"
+		link.Mode = fs.ModeSymlink
+		link.Executable = true
+
+		if err := os.Symlink("a.txt", "a.lnk"); err != nil {
+			t.Fatal(err)
+		}
+	} else {
+		// Windows: create a copy
+		if err := os.WriteFile("a.lnk", []byte("file a"), 0644); err != nil {
+			t.Fatal(err)
+		}
+	}
```

Comment:
Why is this needed? Doesn't GH Action run with privileged user?

Reply:
I wanted tests to be runnable on windows hosts other than CI

Reply:
But then you are not really testing symlink logic on Windows.

Reply:
maybe use `if runtime.GOOS != "windows" && notAdmin {` or something like that.

Reply:
Othrwise you may as well have
```go
if runtime.GOOS != "windows" {
    t.Skip()
}
```
On top of the test.

Reply:
Ok that's little exaggerated, it still checks regular files/dirs.

Reply:
I think I am okay with this.
Resolved.